lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Migrations.hs

```diff
@@ -0,0 +1,481 @@
+                [ expectResponseCode @IO HTTP.status200
+                , expectField (#migrationCost . #getQuantity)
+                    (.> 0)
+                ]
+
+    it "SHELLEY_CALCULATE_02 - \
+        \Cannot calculate fee for empty wallet."
+        $ \ctx -> do
+            w <- emptyWallet ctx
+            let ep = Link.getMigrationInfo @'Shelley w
+            r <- request @ApiWalletMigrationInfo ctx ep Default Empty
+            verify r
+                [ expectResponseCode @IO HTTP.status403
+                , expectErrorMessage (errMsg403NothingToMigrate $ w ^. walletId)
+                ]
+
+    it "SHELLEY_CALCULATE_02 - \
+        \Cannot calculate fee for wallet with dust, that cannot be migrated."
+        $ \ctx -> do
+            -- NOTE
+            -- Special mnemonic for which wallet has dust
+            -- (10 utxo with 33 lovelace)
+            let mnemonics =
+                    ["either", "flip", "maple", "shift", "dismiss", "bridge"
+                    , "sweet", "reveal", "green", "tornado", "need", "patient"
+                    , "wall", "stamp", "pass"] :: [Text]
+            let payloadRestore = Json [json| {
+                    "name": "Dust Shelley Wallet",
+                    "mnemonic_sentence": #{mnemonics},
+                    "passphrase": #{fixturePassphrase}
+                    } |]
+            (_, w) <- unsafeRequest @ApiWallet ctx
+                (Link.postWallet @'Shelley) payloadRestore
+            let ep = Link.getMigrationInfo @'Shelley w
+            r <- request @ApiWalletMigrationInfo ctx ep Default Empty
+            verify r
+                [ expectResponseCode @IO HTTP.status403
+                , expectErrorMessage (errMsg403NothingToMigrate $ w ^. walletId)
+                ]
+
+    describe "SHELLEY_CALCULATE_03 - \
+        \Cannot estimate migration for Byron wallet using Shelley endpoint" $ do
+          forM_ [ ("Byron", emptyRandomWallet)
+                , ("Icarus", emptyIcarusWallet)
+                ] $ \(walType, byronWallet) -> do
+
+                it ("Cannot calculate Shelley migration using wallet: " ++ walType)
+                    $ \ctx -> do
+                    w <- byronWallet ctx
+                    let ep = Link.getMigrationInfo @'Shelley w
+                    r <- request @ApiWalletMigrationInfo ctx ep Default Empty
+                    expectResponseCode @IO HTTP.status404 r
+                    expectErrorMessage (errMsg404NoWallet $ w ^. walletId) r
+
+    describe "SHELLEY_MIGRATE_01 - \
+        \after a migration operation successfully completes, the correct \
+        \amount eventually becomes available in the target wallet for arbitrary \
+        \ number of specified addresses. Balance of source wallet = 0."
+        $ do
+              testAddressCycling 1
+              testAddressCycling 3
+              testAddressCycling 10
+
+    it "SHELLEY_MIGRATE_01 - \
+        \ migrate a big wallet requiring more than one tx" $ \ctx -> do
```

Comment:
This test is failing with:

```
Failures:

  src/Test/Integration/Scenario/API/Shelley/Migrations.hs:221:15: 
  1) API Specifications SHELLEY_MIGRATE_01 -  migrate a big wallet requiring more than one tx
       expected: Status {statusCode = 202, statusMessage = "Accepted"}
        but got: Status {statusCode = 500, statusMessage = "Internal Server Error"}
       
       from the following response: 
Left (DecodeFailure 
"{\"code\":\"created_invalid_transaction\",
  \"message\":\"That's embarrassing. It looks like I've created an invalid transaction that could not be 
    parsed by the node. Here's an error message that may help with debugging: 
    ApplyTxError [LedgerFailure (UtxowFailure (UtxoFailure (MaxTxSizeUTxO 20614 4096)))]\"}")

```

Reply:
Yes. The max number of inputs is currently hard-coded to `100` in the Shelley transaction layer. We need something a bit better here :)

Reply:
Shall I make this test pending and create a ticket for that perhaps?